cats/slf4j/src/main/scala/io/chrisdavenport/log4cats/slf4j/internal/LoggerMacros.scala

```diff
@@ -154,67 +157,85 @@ private[slf4j] object ReflectiveLogMacros {
       }
     }
 
-    msg match {
-      case _ if context.nonEmpty =>
+    (msg, context) match {
+      case (_, Some(ctxExp)) =>
         val MDC = q"org.slf4j.MDC"
-        val Seq = q"scala.collection.Seq"
+        // val Seq = q"scala.collection.Seq"
         val backup = TermName(c.freshName("mdcBackup"))
         q"""
            if ($checkExpr) $F.delay {
              val $backup = $MDC.getCopyOfContextMap
              try {
-               for ((k, v) <- $Seq(..$context)) $MDC.put(k, v)
+               for {
+                 (k, v) <- $ctxExp.toSeq
+                } yield $MDC.put(k, v)
```

Comment:
Do you want the yield here? Desugars to flatmap and map versus foreach without it and since this is done for put side effect only ... I’ve got bitten by weird optimizations around ignored non unit expressions ... likely not related but my two cents from the peanut gallery. :-)

Reply:
I agree that is not right. Also, converting from `Map` to `Seq` is not necessary for iteration.

Reply:
:+1: for `foreach` here

Reply:
Updated back to foreach